helpers/integration-tests/staticUrls.js

```diff
@@ -0,0 +1,45 @@
+const puppeteer = require('puppeteer');
+
+const staticURLsToVerify = [
+    'https://sonarwhal-staging.azurewebsites.net/',
+    'http://pudim.com.br/xunda.jpg',
+    'https://sonarwhal-staging.azurewebsites.net/search/?q=bla',
+    'https://sonarwhal-staging.azurewebsites.net/about/changelog/1',
+    'https://sonarwhal-staging.azurewebsites.net/docs/user-guide/hints/'
+];
+
+const runPuppeteer = async (url) => {
+    const browser = await puppeteer.launch();
+    const page = await browser.newPage();
+    const response = await page.goto(url);
+
+    console.log(
+        `${
+            response.status() === 200 ? '✅ Success' : '❌ Failure'
+        }! ${url} => Status: ${response.status()}`
+    );
+
+    await browser.close();
+
+    return response.status();
+};
+
+const runStaticTests = () => {
+    return new Promise((resolve) => {
+        let errorFound = false;
+
+        staticURLsToVerify.forEach(async (url, index) => {
```

Comment:
I believe `.forEach` doesn't `await` so all the items in the array are called at the same time and thus starting multiple different instances of the browser. When testing this how many separate chrome windows did you see? Was it just one being opened and closed or several?

If you saw multiple instances then we should probably change this to something like this:

```js
const runStaticTests = async () => {
    let errorFound = false;

    for (const url of staticURLsToVerify) {
         const resultStatus = await runPuppeteer(url);
        
        if (resultStatus !== 200) {
            errorFound = true;
        }
    }

    return errorFound;
}
```

Reply:
@molant I'm running the Chrome headless version so I don't observe the windows popping up - thus I can't answer your question about the instances.

When running on the CLI they appear pretty much sequential on all my tests. I'm assuming that if they're being fired sequentially the printing (console.log) order would change on ad hoc basis.

Reply:
As I suspected [`forEach` doesn't `await`](https://stackoverflow.com/questions/37576685/using-async-await-with-a-foreach-loop) , I've been bitten by this in the past 😅

With the code I posted earlier you don't need access to the index. Because the urls are being analyzed sequentially, as soon as the `for..of` ends you can return `errorFound`. We also add the keyword `async` so no need to create a `Promise` object.

Reply:
That's great, thanks @molant 
I will do the fixes later today! 👍

Reply:
@molant Done!